mysql-test/main/select.test

```diff
@@ -4816,3 +4816,13 @@ DROP TABLE t1;
 DROP VIEW v1, v2;
 
 --echo End of 10.0 tests
+
+--echo #
+--echo # Bug MDEV-31116: SIGSEGV in test_if_skip_sort_order|JOIN::optimize_stage2
+--echo #
+SET sql_select_limit=1;
+CREATE TABLE t (a BINARY (2),b BINARY (1),KEY(a));
+INSERT INTO t (a) VALUES (''),(''),(''),(''),(''),(''),(''),(''),(''),(''),
+  (''),(''),(''),('');
+SELECT count(*) FROM t WHERE a IN (SELECT a FROM t WHERE a >'') ORDER BY a;
+DROP TABLE t;
```

Comment:
Convention is when test append to the end of a test case. A marker is made so it be merged easier

```
--echo End of 11.0 tests
```

Reply:
Thank you, I will add this.